edith-step/src/components/ReceiptInput.js

```diff
@@ -1,116 +1,230 @@
 import React from 'react';
 import axios from 'axios';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+import 'regenerator-runtime/runtime';
 
+/**
+ * Displays form that takes in user input for a dynamic number
+ * of grocery items and their prices and quantities and adds them
+ * to a list.
+ */
 export default class ReceiptInput extends React.Component {
+  /**
+   * Constructor
+   * @param {Props} props Setup state.
+   */
   constructor(props) {
     super(props);
     this.state = {items: [], itemName: '', itemPrice: 0.0, itemQuantity: 1};
     this.handleChange = this.handleChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
-  handleSubmit(e) {
+  /**
+   * Sends item info to the receipt-deals servlet and
+   * responds with the cheapest store for that item.
+   * @param {String} name item name
+   * @param {double} price item price
+   * @param {number} quantity item quantity
+   * @return {Object} newDeal cheapest item
+   */
+  async getDeal(name, price, quantity) {
+    const response = await axios({
+      method: 'post',
+      url: '/receipt-deals',
+      data: {
+        itemName: name,
+        itemPrice: price,
+        itemQuantity: quantity,
+      },
+    });
+    const dealItem = response.data;
+
+    const newDeal = dealItem === 'no deal found' ?
+      {storeName: 'no deal found', storePrice: 0} :
+      {storeName: dealItem.store, storePrice: dealItem.price};
+
+    return newDeal;
+  }
+
+  /**
+   * Send a post request to the receipt-data servlet
+   * with a grocery item in it every time a new item
+   * is added.
+   * @param {Event} e Submission event.
+   */
+  async handleSubmit(e) {
     e.preventDefault();
     if (this.state.itemName.length === 0) {
       return;
     }
+
+    const newDeal = await this.getDeal(this.state.itemName,
+        this.state.itemPrice, this.state.itemQuantity);
+    let dealMessage;
+    if (newDeal.storeName == 'no deal found' ||
+        newDeal.storePrice > this.state.itemPrice) {
+      dealMessage = 'no deal found';
+    } else {
+      dealMessage = 'Purchase at ' +
+      newDeal.storeName + ' for $' +
+      newDeal.storePrice + '.';
+    }
+
     const newItem = {
       itemName: this.state.itemName,
       itemPrice: this.state.itemPrice,
       itemQuantity: this.state.itemQuantity,
-      id: Date.now()
+      itemDeal: dealMessage,
+      id: Date.now(),
     };
-    
+
     axios({
       method: 'post',
       url: '/receipt-data',
       data: {
         itemName: this.state.itemName,
         itemPrice: this.state.itemPrice,
-        itemQuantity: this.state.itemQuantity
-      }
-    }).then((response) => {
-      console.log(response);
+        itemQuantity: this.state.itemQuantity,
+      },
     });
 
-    this.setState(state => ({
+    this.setState((state) => ({
       items: state.items.concat(newItem),
```

Comment:
what do you think of using ```...``` spread operator for updating arrays intead ?
https://stackoverflow.com/a/26254086/10688048